core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/AsyncRetryableStage.java

```diff
@@ -200,6 +202,14 @@ private void beforeExecute() {
 
             // Before each attempt, Modify the context to use original request body provider
             context.requestProvider(originalRequestBody);
+
+            // For streaming requests, future from the execution attributes (set in BaseAsyncClientHandler#execute method)
+            // should be used on first attempt. For all retry attempts, clear the value so that prepare() method is
+            // called (in MakeAsyncHttpRequestStage) to get a new future
+            if (requestCount > 1 && context.streamingAsyncResponseTransformerFuture() != null) {
+                context.executionAttributes().putAttribute(STREAMING_ASYNC_RESPONSE_TRANSFORMER_FUTURE, null);
```

Comment:
Is there any reason we are not adding `asyncResponseTransformerFuture(CompletableFuture<?> completableFuture)` in `RequestExecutionContext`?

Reply:
No particular reason. There were not lot of methods in the class that are modifying the state, so didn't add new method for this.